### DISCUSSION TITLE
What is the use of Bearer Token and x-authenticated-user token?

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Ed/Community#249] [Documentation] [answered]
We would like to understand in SB portal what is use of Bearer Token and X-authenticated-user token.
How do we decide which APIs / service we need to use either/both of them and when accessing APIs via postman, we have noticed for some APIs we need to provide Bearer Token and for some other APIs it requires x-authenticated-user token also. Can you please clarify these?

cc: @reshmi-nair @rajeevsathish @Krishnaj20

**Comment by Krishnaj20** [comment, marked as answer]
@sharathkashyap - Pls look into this.

**Reply by beepdot** [reply]
Bearer token - This is used to cross the API gateway layer, which is Kong in Sunbird. This ensures that you are allowed to use this particular API.
X Auth Token - This is a user specific token, issued by Keycloak, which will have info about the user like user id, when the user logged in etc. This is used to perform checks and validations at a user level such as do you have the access to create contents (the user needs to be assigned with a content creator role to do this)

**Comment by chitranshu-keshav** [comment]
@nidhikachhawa24 Can you please mark this thread as answered if your query is resolved?
